bt2_values_generator.rb

```diff
@@ -41,10 +42,22 @@ def get(name, val)
       bt_type = self.class.instance_variable_get(:@bt_type)
       bt_type_is = self.class.instance_variable_get(:@bt_type_is)
 
-      pr "if (#{val} && #{bt_type_is}(#{val}))"
+      pr "if (#{val} != NULL) {"
+      pr "  if (!#{bt_type_is}(#{val})) {"
+      pr "    printf(\"Bad value for command line argument '%s' the value must be %s \\n\",\"#{@name}\",\"#{bt_type}\");"
+      pr "    exit(1);"
+      pr "  }"
       pr "  #{name} = #{cast_func}bt_value_#{bt_type}_get(#{val});"
-      pr 'else'
-      pr "  #{name} = #{cast_func}#{bt_default_value};"
+      if @default_value != nil
+        pr "} else {"
+        pr "  #{name} = #{cast_func}#{ default_value };"
+        pr "}"
+      else
+        pr "} else {"
+        pr "  #{name} = #{cast_func}#{bt_default_value};"
```

Comment:
Pleae do like this`usr_default_value` as arguments.
And then `default_value =  usr_default_value || bt_default_value`
and do `#{name} = #{cast_func}#{default_value};`
So you don't need to have a special case in the code generation.